ui/app/AppLayouts/Onboarding2/pages/KeycardEnterPinPage.qml

```diff
@@ -0,0 +1,171 @@
+
+import AppLayouts.Onboarding2.controls 1.0
+
+import utils 1.0
+
+KeycardBasePage {
+    id: root
+
+    property var tryToSetPinFunction: (pin) => { console.error("tryToSetPinFunction: IMPLEMENT ME"); return false }
+    required property int remainingAttempts
+    property bool unlockUsingSeedphrase
+
+    signal keycardPinEntered(string pin)
+    signal reloadKeycardRequested()
+    signal unlockWithSeedphraseRequested()
+    signal keycardFactoryResetRequested()
+
+    pageClassName: "KeycardEnterPinPage"
+    image.source: Theme.png("onboarding/keycard/reading")
+
+    QtObject {
+        id: d
+        property string tempPin
+        property bool pinValid
+    }
+
+    buttons: [
+        StatusPinInput {
+            id: pinInput
+            anchors.horizontalCenter: parent.horizontalCenter
+            validator: StatusIntValidator { bottom: 0; top: 999999 }
+            onPinInputChanged: {
+                if (pinInput.pinInput.length === pinInput.pinLen) { // we have the full length PIN now
+                    d.tempPin = pinInput.pinInput
+                    d.pinValid = root.tryToSetPinFunction(d.tempPin)
+                    if (!d.pinValid) {
+                        pinInput.statesInitialization()
+                    }
+                }
+            }
+        },
+        StatusBaseText {
+            id: errorText
+            anchors.horizontalCenter: parent.horizontalCenter
+            text: qsTr("%n attempt(s) remaining", "", root.remainingAttempts)
+            font.pixelSize: Theme.tertiaryTextFontSize
+            color: Theme.palette.dangerColor1
+            visible: false
+        },
+        StatusButton {
+            id: btnFactoryReset
+            width: 320
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.topMargin: Theme.halfPadding
+            visible: false
+            text: qsTr("Factory reset Keycard")
+            onClicked: root.keycardFactoryResetRequested()
+        },
+        StatusButton {
+            id: btnUnlockWithSeedphrase
+            visible: false
+            text: qsTr("Unlock with recovery phrase")
+            anchors.horizontalCenter: parent.horizontalCenter
+            onClicked: root.unlockWithSeedphraseRequested()
+        },
+        StatusButton {
+            id: btnReload
+            width: 320
+            anchors.horizontalCenter: parent.horizontalCenter
+            visible: false
+            text: qsTr("I’ve inserted a different Keycard")
+            normalColor: "transparent"
+            borderWidth: 1
+            borderColor: Theme.palette.baseColor2
+            onClicked: root.reloadKeycardRequested()
+        }
+    ]
+
+    state: "entering"
+
+    states: [
+        State {
+            name: "locked"
+            when: root.remainingAttempts <= 0
+            PropertyChanges {
+                target: root
+                title: "<font color='%1'>".arg(Theme.palette.dangerColor1) + qsTr("Keycard locked") + "</font>"
```

Comment:
Same here with regards to the color. I won't comment everywhere, I guess if you choose to address it, do a CTRL+SHIFT+F :smile: